src/Illuminate/Database/Console/Migrations/MigrateCommand.php

```diff
@@ -163,26 +164,42 @@ protected function repositoryExists()
     {
         return retry(2, fn () => $this->migrator->repositoryExists(), 0, function ($e) {
             try {
-                if ($e->getPrevious() instanceof SQLiteDatabaseDoesNotExistException) {
-                    return $this->createMissingSqliteDatabase($e->getPrevious()->path);
-                }
-
-                $connection = $this->migrator->resolveConnection($this->option('database'));
-
-                if (
-                    $e->getPrevious() instanceof PDOException &&
-                    $e->getPrevious()->getCode() === 1049 &&
-                    in_array($connection->getDriverName(), ['mysql', 'mariadb'])) {
-                    return $this->createMissingMysqlDatabase($connection);
-                }
-
-                return false;
+                return $this->handleMissingDatabase($e);
             } catch (Throwable) {
                 return false;
             }
         });
     }
 
+    /**
+     * Attempt to create the database if it's missing.
+     */
+    protected function handleMissingDatabase(Throwable $e): bool
+    {
+        if ($e->getPrevious() instanceof SQLiteDatabaseDoesNotExistException) {
+            return $this->createMissingSqliteDatabase($e->getPrevious()->path);
+        }
+
+        $connection = $this->migrator->resolveConnection($this->option('database'));
+
+        if (! $e->getPrevious() instanceof PDOException) {
+            return false;
+        }
+
+        if (
+            ($e->getPrevious()->getCode() === 1049 && in_array($connection->getDriverName(), ['mysql', 'mariadb']))
+            || (
+                ($e->getPrevious()->errorInfo[0] ?? null) == '08006'
+                && $connection->getDriverName() == 'pgsql'
+                && Str::contains($e->getPrevious()->getMessage(), '"'.$connection->getDatabaseName().'"')
```

Comment:
Pgsql has no specific code for missing database. https://www.postgresql.org/docs/current/errcodes-appendix.html

<img width="1153" alt="image" src="https://github.com/user-attachments/assets/1028f159-3e36-45c4-8c65-a2dc1e7bd3fc" />

If we don't check for a string such as the database name which is part of the error, a wrong username or password would also offer the user the option to create a database.